src/llfs/volume.test.cpp

```diff
@@ -2079,4 +2118,345 @@ TEST_F(VolumeTest, AppendableJobGrantSize)
   EXPECT_EQ(llfs::packed_sizeof_slot(*packed_commit_job), commit_slot_size);
 }
 
+//==#==========+==+=+=++=+++++++++++-+-+--+----- --- -- -  -  -   -
+//
+TEST_F(VolumeSimTest, DeadPageRefCountSimulation)
+{
+  const u32 max_seeds = batt::getenv_as<int>("MAX_SEEDS").value_or(1000);
+  const u32 yield_count = batt::getenv_as<int>("YIELD_COUNT").value_or(37);
+
+  LOG(INFO) << "Starting DeadPageRefCount test for " << max_seeds << " iterations...";
+  for (u32 current_seed = 0; current_seed < max_seeds; ++current_seed) {
+    LOG_EVERY_N(INFO, 100) << BATT_INSPECT(current_seed) << BATT_INSPECT(max_seeds)
+                           << BATT_INSPECT(yield_count);
+    ASSERT_NO_FATAL_FAILURE(this->run_dead_page_recovery_test(current_seed, yield_count));
+  }
+}
+
+using DeathTestVolumeSimTest = VolumeSimTest;
+
+//==#==========+==+=+=++=+++++++++++-+-+--+----- --- -- -  -  -   -
+//
+TEST_F(DeathTestVolumeSimTest, DeadPageRefCountVariantSimulation)
+{
+  const u32 max_seeds = batt::getenv_as<int>("MAX_SEEDS").value_or(1000);
+
+  // Note that this test fails few times times when pre_halt and post_halt yield values are around
+  // 50 and 40 respectively. To create a perfect failing test use 50 and 40 as the pre_halt and
+  // post_hatl values respectively. .
+
+  LOG(INFO) << "Starting DeadPageRefCountVariant test for " << max_seeds << " iterations...";
+
+  auto main_test_block = [&]() {
+    for (u32 current_seed = 0; current_seed < max_seeds; ++current_seed) {
+      LOG_EVERY_N(INFO, 100) << BATT_INSPECT(current_seed) << BATT_INSPECT(max_seeds);
+      ASSERT_NO_FATAL_FAILURE(
```

Comment:
My goal was to positively hit the issue a few times. Just using seed values gave the right setting to hit the issue in a simple way. Once we hit the issue that's done I added some variability using simulation random scheduling (which is based on seed value). So the test centers around the scenario and goes little off focus from there.  

With RNG approach we will effectively get similar results as the sequence of random yield-counts will be still fixed. It will be one of the sequences of yield-counts. To me this is same as just picking yield-count in increasing order (which is based on current seed value). 

I am not sure if I am missing something here. We can discuss this further during our meeting if you would like.